examples/contrib/detection/faster_rcnn.py

```diff
@@ -0,0 +1,286 @@
+import os
+import random
+import sys
+from argparse import ArgumentParser
+from typing import Any, Tuple
+
+import aim
+import albumentations as A
+import numpy as np
+import torch
+from aim.pytorch_ignite import AimLogger
+from albumentations.pytorch.transforms import ToTensorV2
+from mean_ap import CocoMetric, convert_to_coco_api
+from PIL import Image
+from torch.cuda.amp import GradScaler
+from torch.optim import SGD
+from torch.optim.lr_scheduler import OneCycleLR
+from torch.utils.data import DataLoader, Subset
+from torchvision.datasets import VOCDetection
+from torchvision.models.detection import fasterrcnn_resnet50_fpn
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
+from torchvision.utils import draw_bounding_boxes
+
+from ignite.contrib.handlers.tqdm_logger import ProgressBar
+from ignite.engine import Engine
+from ignite.engine.events import Events
+from ignite.handlers import Checkpoint, DiskSaver, global_step_from_engine
+from ignite.metrics.running_average import RunningAverage
+
+try:
+    from defusedxml.ElementTree import parse as ET_parse
+except ImportError:
+    from xml.etree.ElementTree import parse as ET_parse
+
+
+class Dataset(VOCDetection):
+
+    classes = (
+        "aeroplane",
+        "bicycle",
+        "boat",
+        "bus",
+        "car",
+        "motorbike",
+        "train",
+        "bottle",
+        "chair",
+        "diningtable",
+        "pottedplant",
+        "sofa",
+        "tvmonitor",
+        "bird",
+        "cat",
+        "cow",
+        "dog",
+        "horse",
+        "sheep",
+        "person",
+    )
+    name2class = {v: k + 1 for k, v in enumerate(classes)}
+    class2name = {k + 1: v for k, v in enumerate(classes)}
+
+    def __getitem__(self, index: int) -> Tuple[Any, Any]:
```

Comment:
It seems that what is done here could be implemented as a transformation. It should be moved in another file IMO.

Reply:
Do you mean the name and class number conversion? If yes I think it's ok to place here since there are nothing to do with the images.